Sprint-3/implement/is-proper-fraction.test.js

```diff
@@ -0,0 +1,88 @@
+// You wil need to implement a function isProperFraction
+// You need to write assertions for your function to check it works in different cases
+
+// Terms:
+// Fractions: https://www.bbc.co.uk/bitesize/topics/zt9n6g8/articles/zjxpp4j
+// Written here like this: 1/2 == Numerator/Denominator
+
+// Acceptance criteria:
+
+// Proper Fraction check:
+// Input: numerator = 2, denominator = 3
+// target output: true
+// Explanation: The fraction 2/3 is a proper fraction, where the numerator is less than the denominator. The function should return true.
+
+// Improper Fraction check:
+// Input: numerator = 5, denominator = 2
+// target output: false
+// Explanation: The fraction 5/2 is an improper fraction, where the numerator is greater than or equal to the denominator. The function should return false.
+
+// Zero Denominator check:
+// Input: numerator = 3, denominator = 0
+// No target output: Error (Denominator cannot be zero)
+// Explanation: The function should throw an error when the denominator is zero, as it's not a valid fraction.
+
+// Negative Fraction check:
+// Input: numerator = -4, denominator = 7
+// target output: true
+// Explanation: The fraction -4/7 is a proper fraction because the absolute value of the numerator (4) is less than the denominator (7). The function should return true.
+
+// Equal Numerator and Denominator check:
+// Input: numerator = 3, denominator = 3
+// target output: false
+// Explanation: The fraction 3/3 is not a proper fraction because the numerator is equal to the denominator. The function should return false.
+// These acceptance criteria cover a range of scenarios to ensure that the isProperFraction function handles both proper and improper fractions correctly and handles potential errors such as a zero denominator.
+
+function isProperFraction(numerator, denominator){
```

Comment:
Typically, we would write the function in the file is-proper-fraction.js and the test in the paired filed is-proper-fraction.test.js

Then we would import the function into the test file to make it available to the tests (just like in html we import the css file to make it work on the html).

I'm not sure this was completely clear in the coursework; what do you think? How could we revise it to make it clearer?

Reply:
Yes, I started to create separate files for tests and functions from the next sprint after this. It's really more clear.